File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/Converters/ProductObjectConverter.vb
Imports System.Globalization
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data

<ValueConversion(GetType(Product), GetType(String))> _
   Public Class ProductObjectConverter
  Implements IValueConverter

  Public Function Convert(ByVal value As Object, _
                          ByVal targetType As Type, _
                          ByVal parameter As Object, _
                          ByVal culture As CultureInfo) As Object _
                 Implements System.Windows.Data.IValueConverter.Convert
    Dim prod As Product
    Dim ret As String = String.Empty

    prod = DirectCast(value, Product)
    If prod IsNot Nothing Then
      ret = (prod.ProductName & " (ProductId=") + prod.ProductId.ToString() & ")"
    End If

    Return ret
  End Function

  Public Function ConvertBack(ByVal value As Object, _
                              ByVal targetType As Type, _
                              ByVal parameter As Object, _
                              ByVal culture As CultureInfo) As Object _
                  Implements System.Windows.Data.IValueConverter.ConvertBack
    Return System.Convert.ToString(value)
  End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/Converters/ProductObjectConverter.cs
using System;
using System.ComponentModel;
using System.Globalization;
using System.Windows.Data;

namespace WPFComponents
{
  [ValueConversion(typeof(Product), typeof(string))]
  public class ProductObjectConverter : IValueConverter
  {
    public object Convert(object value, Type targetType,
                                    object parameter, CultureInfo culture)
    {
      Product prod;
      string ret = string.Empty;

      prod = (Product)value;
      if(prod != null)
        ret = prod.ProductName + "  (ProductId=" + prod.ProductId.ToString() + ")";

      return ret;
    }

    public object ConvertBack(object value, Type targetType,
                                                object parameter, CultureInfo culture)
    {
      return System.Convert.ToString(value);
    }
  }
}
